src/java.net.http/share/classes/jdk/internal/net/http/ExchangeImpl.java

```diff
@@ -58,6 +58,8 @@ abstract class ExchangeImpl<T> {
 
     final Exchange<T> exchange;
 
+    volatile boolean expectTimeoutRaised;
```

Comment:
Could you make this field private and add a method to het it instead? For instance 
`final boolean expectTimeoutRaised() { return expectTimeoutRaised;}`